### DISCUSSION TITLE
How to convert bmp file from 24-bit color depth to 16-bit?

### DISCUSSION THREAD

**Discussion by yungchunw** [ImageMagick/ImageMagick#5621] [Help]
Trying to convert color depth from 24-bit to 16-bit with ImageMagick Convert.exe. But the result is not my expected.

Command
```
convert '24_bit.bmp' -alpha off -define bmp:subtype=RGB565 '16_bit.bmp'
```

BMP Hex Compare

(A) result is the output from Imagemagick convert 

(B) result is waht I expected

![](https://i.stack.imgur.com/RoJPA.png)

**Comment by snibgo** [comment]
Why did you you expect it to be different?

Two programs may create different files, but both files may be correct. They may represent the same image, and both comply to standards.

So, do you think the IM result is not correct?

**Reply by yungchunw** [reply]
Thank for your reply~
I know both of them are look same and standards. And the question I ask  because I have to covert these 16-bit bmp files to a .spr (Sprite image file) for MMRPG Game.

More details explame here:

- ViX Tool : https://ja.wikipedia.org/wiki/ViX

![](https://imgur.com/7vlfG0o.png)

**Comment by snibgo** [comment]
What is your version of IM? On what platform?

If your IM is old, perhaps "`-define bmp:subtype=RGB565`" wasn't implemented.

What is your input image?

**Reply by yungchunw** [reply]
Version: ImageMagick-7.1.0-portable-Q16-HDRI-x64
Plateform: Win11
Input image: 24-bit bmp file ([download link](https://www.sendspace.com/file/zc6pbm))

**Comment by snibgo** [comment]
Gimp and Windows Photo Viewer and IM agree that your input 257-0-000.bmp looks like this:
![x](https://user-images.githubusercontent.com/33812368/194119179-3b7442a5-102f-4c7e-8380-3fd5129b6d60.png)

This is the same as the thumbnail in your images, but different to both "Result in Game" images.

**Reply by yungchunw** [reply]
Yes, it's right. Because red background will been transparent.

**Reply by yungchunw** [reply]
https://stackoverflow.com/questions/73955424/imagemagick-convert-bmp-from-24-bit-to-16-bit-question
